pkg/resource/computed_attributes.go

```diff
@@ -0,0 +1,184 @@
+package resource
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/seal-io/walrus/pkg/dao/model"
+	"github.com/seal-io/walrus/pkg/dao/model/environment"
+	"github.com/seal-io/walrus/pkg/dao/model/project"
+	"github.com/seal-io/walrus/pkg/dao/model/resourcedefinitionmatchingrule"
+	"github.com/seal-io/walrus/pkg/dao/model/templateversion"
+	"github.com/seal-io/walrus/pkg/dao/types"
+	"github.com/seal-io/walrus/pkg/dao/types/property"
+	pkgenv "github.com/seal-io/walrus/pkg/environment"
+	"github.com/seal-io/walrus/utils/json"
+)
+
+// GenComputedAttributes generate computed attributes for resource.
+// Required:
+// entity.EnvironmentID, entity.Attributes
+// entity.TemplateID/entity.ResourceDefinitionMatchingRuleID.
+func GenComputedAttributes(
+	ctx context.Context,
+	entity *model.Resource,
+	modelClient *model.Client,
```

Comment:
we follow the pattern `func name(context.Context, model.ClientSet`, should we change the order of the params? The modelClient is `*model.Client`, will the function call pass a transaction model client?